dotcom-rendering/src/components/EmailSignUpSwitcher.tsx

```diff
@@ -0,0 +1,30 @@
+import { AppEmailSignUp } from './AppEmailSignUp.importable';
+import { useConfig } from './ConfigContext';
+import type { EmailSignUpProps } from './EmailSignup';
+import { Island } from './Island';
+import { WebEmailSignUp } from './WebEmailSignup';
+
+interface EmailSignUpSwitcherProps extends EmailSignUpProps {
+	index: number;
+	identityName: string;
+	successDescription: string;
+	/** You should only set this to true if the privacy message will be shown elsewhere on the page */
+	hidePrivacyMessage?: boolean;
+}
+
+export const EmailSignUpSwitcher = ({
+	index,
+	...emailSignUpProps
+}: EmailSignUpSwitcherProps) => {
+	const { renderingTarget } = useConfig();
```

Comment:
We use config here which is why this switcher is in its own component, rather than being kept in the renderElement file